tsdb/engine/tsm1/engine.go

```diff
@@ -1161,6 +1161,12 @@ func (e *Engine) createVarRefIterator(opt influxql.IteratorOptions, aggregate bo
 				return err
 			}
 
+			if opt.Limit > 0 || opt.Offset > 0 || opt.SLimit > 0 || opt.SOffset > 0 {
```

Comment:
Tag sets are ordered by the returned series. Internal to a series wouldn't be ordered so if you had this:

```
cpu,host=server01,region=uswest value=0
cpu,host=server01,region=useast value=1
cpu,host=server02 value=2

> SELECT * FROM cpu GROUP BY host
```

The host would be ordered, but the internal series grouped into the `server01` bucket would not be deterministic. This is why it affects `LIMIT <n>` and `OFFSET <n>`. While I'm not sure it matters, I assume we still want that to be deterministic.